In [5]:
import os
import pandas as pd
from datetime import datetime

In [2]:
path="Weekly Files Not Duplicated"

In [3]:
dir_list = os.listdir(path)
dir_list

['Linkedin_Job_Links_03-Oct-2023-to-09-Oct-2023_1875-jobs_89-pages.csv',
 'Linkedin_Job_Links_07-Nov-2023-to-13-Nov-2023_754-jobs_32-pages.csv',
 'Linkedin_Job_Links_10-Oct-2023-to-16-Oct-2023_1928-jobs_93-pages.csv',
 'Linkedin_Job_Links_14-Nov-2023-to-20-Nov-2023_1560-jobs_70-pages.csv',
 'Linkedin_Job_Links_17-Oct-2023-to-23-0ct-2023_1844-jobs_97-pages.csv',
 'Linkedin_Job_Links_21-Nov-2023-to-27-Nov-2023_1036-jobs_57-pages.csv',
 'Linkedin_Job_Links_24-Oct-2023-to-30-0ct-2023_1824-jobs_83-pages.csv',
 'Linkedin_Job_Links_26-Sep-2023-to-02-Oct-2023_1912-jobs_92-pages.csv',
 'Linkedin_Job_Links_31-Oct-2023-to-06-Nov-2023_1561-jobs_72-pages.csv']

In [4]:
len(dir_list)

9

In [11]:
dates=[file.split('_')[3].split('-to-')[0] for file in dir_list]
dates.sort(key=lambda date: datetime.strptime(date, "%d-%b-%Y"))
file_names=[]
for date in dates:
    for file in dir_list:
        if file.split('_')[3].split('-to-')[0]==date:
            file_names.append(file)
file_names    

['Linkedin_Job_Links_26-Sep-2023-to-02-Oct-2023_1912-jobs_92-pages.csv',
 'Linkedin_Job_Links_03-Oct-2023-to-09-Oct-2023_1875-jobs_89-pages.csv',
 'Linkedin_Job_Links_10-Oct-2023-to-16-Oct-2023_1928-jobs_93-pages.csv',
 'Linkedin_Job_Links_17-Oct-2023-to-23-0ct-2023_1844-jobs_97-pages.csv',
 'Linkedin_Job_Links_24-Oct-2023-to-30-0ct-2023_1824-jobs_83-pages.csv',
 'Linkedin_Job_Links_31-Oct-2023-to-06-Nov-2023_1561-jobs_72-pages.csv',
 'Linkedin_Job_Links_07-Nov-2023-to-13-Nov-2023_754-jobs_32-pages.csv',
 'Linkedin_Job_Links_14-Nov-2023-to-20-Nov-2023_1560-jobs_70-pages.csv',
 'Linkedin_Job_Links_21-Nov-2023-to-27-Nov-2023_1036-jobs_57-pages.csv']

In [13]:
file_names[:7]

['Linkedin_Job_Links_26-Sep-2023-to-02-Oct-2023_1912-jobs_92-pages.csv',
 'Linkedin_Job_Links_03-Oct-2023-to-09-Oct-2023_1875-jobs_89-pages.csv',
 'Linkedin_Job_Links_10-Oct-2023-to-16-Oct-2023_1928-jobs_93-pages.csv',
 'Linkedin_Job_Links_17-Oct-2023-to-23-0ct-2023_1844-jobs_97-pages.csv',
 'Linkedin_Job_Links_24-Oct-2023-to-30-0ct-2023_1824-jobs_83-pages.csv',
 'Linkedin_Job_Links_31-Oct-2023-to-06-Nov-2023_1561-jobs_72-pages.csv',
 'Linkedin_Job_Links_07-Nov-2023-to-13-Nov-2023_754-jobs_32-pages.csv']

In [14]:
df_concat=None
for file in file_names[:7]:
    df=pd.read_csv(path+'/'+file)
    df_concat=pd.concat([df_concat,df],ignore_index=True)    

In [15]:
df_concat

Job Title  \
0              LinkedIn Data Extractor   
1                 It Management Intern   
2               SQL Database Developer   
3      Business Intelligence Developer   
4           Web Developer - Internship   
...                                ...   
11693       Sr. Software Engineer - UI   
11694    Customer Sales Representative   
11695           Telesales Agent or CSR   
11696                  Sales Executive   
11697                Angular Developer   

                                                Job Link      Job ID  
0      https://www.linkedin.com/jobs/view/3727054368/...  3727054368  
1      https://www.linkedin.com/jobs/view/3723395858/...  3723395858  
2      https://www.linkedin.com/jobs/view/3727034665/...  3727034665  
3      https://www.linkedin.com/jobs/view/3727029273/...  3727029273  
4      https://www.linkedin.com/jobs/view/3727034696/...  3727034696  
...                                                  ...         ...  
11693  https://www.linkedin.com/jobs/view/3758268626/...  3758268626  
11694  https://www.linkedin.com/jobs/view/3753070889/...  3753070889  
11695  https://www.linkedin.com/jobs/view/3757800035/...  3757800035  
11696  https://www.linkedin.com/jobs/view/3757283989/...  3757283989  
11697  https://www.linkedin.com/jobs/view/3751461639/...  3751461639  

[11698 rows x 3 columns]

In [16]:
total_pages=0
for file in file_names:
    total_pages=total_pages+int(file.split('_')[5].strip('-pages.csv'))
total_pages

685

In [17]:
total_records=0
for file in file_names:
    total_records=total_records+int(file.split('_')[4].strip('-jobs'))
total_records

14294

In [18]:
len(df_concat)

11698

### checking data duplication

In [6]:
len(df_concat['Job ID'].unique())

NameError: name 'df_concat' is not defined

In [24]:
duplicates_count=df_concat.duplicated(subset=['Job ID']).sum()
duplicates_count

1490

In [26]:
duplicates = df_concat['Job ID'].duplicated()
df_concat[duplicates]

Job Title  \
1917                                  Senior GRC Analyst   
1923   Technology Consulting- SAP- EAM- Rapid Innovation   
1924                        Pre-Sales Solutions Engineer   
1931                                     DevOps Engineer   
1933   Business Development Executive (REMOTE) up to ...   
...                                                  ...   
11641                       Software Engineering Manager   
11645                                   Media Generalist   
11665                         Talent Acquisition Manager   
11669                                     Security Guard   
11685                            Sales Account Executive   

                                                Job Link      Job ID  
1917   https://www.linkedin.com/jobs/view/3705698711/...  3705698711  
1923   https://www.linkedin.com/jobs/view/3637936648/...  3637936648  
1924   https://www.linkedin.com/jobs/view/3722491418/...  3722491418  
1931   https://www.linkedin.com/jobs/view/3730356736/...  3730356736  
1933   https://www.linkedin.com/jobs/view/3728108177/...  3728108177  
...                                                  ...         ...  
11641  https://www.linkedin.com/jobs/view/3737888915/...  3737888915  
11645  https://www.linkedin.com/jobs/view/3755745473/...  3755745473  
11665  https://www.linkedin.com/jobs/view/3742000837/...  3742000837  
11669  https://www.linkedin.com/jobs/view/3755630853/...  3755630853  
11685  https://www.linkedin.com/jobs/view/3756313333/...  3756313333  

[1490 rows x 3 columns]

In [27]:
df_no_duplicates = df_concat.drop_duplicates(subset=['Job ID'])
df_no_duplicates

Job Title  \
0              LinkedIn Data Extractor   
1                 It Management Intern   
2               SQL Database Developer   
3      Business Intelligence Developer   
4           Web Developer - Internship   
...                                ...   
11693       Sr. Software Engineer - UI   
11694    Customer Sales Representative   
11695           Telesales Agent or CSR   
11696                  Sales Executive   
11697                Angular Developer   

                                                Job Link      Job ID  
0      https://www.linkedin.com/jobs/view/3727054368/...  3727054368  
1      https://www.linkedin.com/jobs/view/3723395858/...  3723395858  
2      https://www.linkedin.com/jobs/view/3727034665/...  3727034665  
3      https://www.linkedin.com/jobs/view/3727029273/...  3727029273  
4      https://www.linkedin.com/jobs/view/3727034696/...  3727034696  
...                                                  ...         ...  
11693  https://www.linkedin.com/jobs/view/3758268626/...  3758268626  
11694  https://www.linkedin.com/jobs/view/3753070889/...  3753070889  
11695  https://www.linkedin.com/jobs/view/3757800035/...  3757800035  
11696  https://www.linkedin.com/jobs/view/3757283989/...  3757283989  
11697  https://www.linkedin.com/jobs/view/3751461639/...  3751461639  

[10208 rows x 3 columns]

In [28]:
len(df_no_duplicates)

10208

In [29]:
len(df_concat)-duplicates_count

10208

## toatal non duplicated records till 7th week:10208

### saving file

In [30]:
folder_path='one file from all weeks/'
folder_path

'one file from all weeks/'

In [31]:
week='26-Sep-2023-to-13-Nov-2023'
week

'26-Sep-2023-to-13-Nov-2023'

In [32]:
df_no_duplicates.to_csv(folder_path+f'Linkedin_Job_Links_{week}_{len(df_no_duplicates)}-jobs_{total_pages}-pages_not_duplicated.csv',index=False)

### concatenating Daily Files Not Duplicated

In [17]:
path2="Daily Files Not Duplicated/"

In [18]:
dir_list = os.listdir(path2)
dir_list

['1st Week 26 Sep-02 Oct',
 '2nd Week 03 Oct-09 Oct',
 '3rd week 10 Oct -16 0ct',
 '4th week 17 Oct -23 0ct',
 '5th week 24 Oct -30 0ct',
 '6th week 31 Oct -06 Nov',
 '7th week 07 Nov -13 Nov',
 '8th week 14 Nov -20 Nov',
 '9th week 21 Nov -27 Nov']

In [19]:
List_of_csv_files=[]
List_of_csv_file_paths=[]
for dir in dir_list[:]:
    # print(os.listdir(path2+dir))
    dir_list2=os.listdir(path2+dir)
    paths=[]
    for dir2 in dir_list2:
        path=path2+dir+'/'+dir2
        paths.append(path)
    dates=[file.split('/')[2].split('_')[3] for file in paths]
    dates.sort(key=lambda date: datetime.strptime(date, "%d-%b-%Y"))
    paths2=[]
    for date in dates:
        for file in paths:
            if file.split('/')[2].split('_')[3]==date:
                paths2.append(file)

        
    List_of_csv_files.extend(dir_list2)
    List_of_csv_file_paths.extend(paths2)

In [20]:
List_of_csv_files

['Linkedin_Job_Links_01-Oct-2023_125-jobs_5-pages.csv',
 'Linkedin_Job_Links_02-Oct-2023_360-jobs_15-pages.csv',
 'Linkedin_Job_Links_26-Sep-2023_561-jobs_23-pages.csv',
 'Linkedin_Job_Links_27-Sep-2023_352-jobs_18-pages.csv',
 'Linkedin_Job_Links_28-Sep-2023_309-jobs_14-pages.csv',
 'Linkedin_Job_Links_29-Sep-2023_239-jobs_10-pages.csv',
 'Linkedin_Job_Links_30-Sep-2023_170-jobs_7-pages.csv',
 'Linkedin_Job_Links_03-Oct-2023_375-jobs_15-pages.csv',
 'Linkedin_Job_Links_04-Oct-2023_356-jobs_15-pages.csv',
 'Linkedin_Job_Links_05-Oct-2023_335-jobs_15-pages.csv',
 'Linkedin_Job_Links_06-Oct-2023_332-jobs_14-pages.csv',
 'Linkedin_Job_Links_07-Oct-2023_250-jobs_11-pages.csv',
 'Linkedin_Job_Links_08-Oct-2023_116-jobs_5-pages.csv',
 'Linkedin_Job_Links_09-Oct-2023_334-jobs_14-pages.csv',
 'Linkedin_Job_Links_10-Oct-2023_421-jobs_18-pages.csv',
 'Linkedin_Job_Links_11-Oct-2023_374-jobs_17-pages.csv',
 'Linkedin_Job_Links_12-Oct-2023_261-jobs_12-pages.csv',
 'Linkedin_Job_Links_13-Oct-2023_3

In [21]:
len(List_of_csv_files)

43

In [22]:
List_of_csv_file_paths

['Daily Files Not Duplicated/1st Week 26 Sep-02 Oct/Linkedin_Job_Links_26-Sep-2023_561-jobs_23-pages.csv',
 'Daily Files Not Duplicated/1st Week 26 Sep-02 Oct/Linkedin_Job_Links_27-Sep-2023_352-jobs_18-pages.csv',
 'Daily Files Not Duplicated/1st Week 26 Sep-02 Oct/Linkedin_Job_Links_28-Sep-2023_309-jobs_14-pages.csv',
 'Daily Files Not Duplicated/1st Week 26 Sep-02 Oct/Linkedin_Job_Links_29-Sep-2023_239-jobs_10-pages.csv',
 'Daily Files Not Duplicated/1st Week 26 Sep-02 Oct/Linkedin_Job_Links_30-Sep-2023_170-jobs_7-pages.csv',
 'Daily Files Not Duplicated/1st Week 26 Sep-02 Oct/Linkedin_Job_Links_01-Oct-2023_125-jobs_5-pages.csv',
 'Daily Files Not Duplicated/1st Week 26 Sep-02 Oct/Linkedin_Job_Links_02-Oct-2023_360-jobs_15-pages.csv',
 'Daily Files Not Duplicated/2nd Week 03 Oct-09 Oct/Linkedin_Job_Links_03-Oct-2023_375-jobs_15-pages.csv',
 'Daily Files Not Duplicated/2nd Week 03 Oct-09 Oct/Linkedin_Job_Links_04-Oct-2023_356-jobs_15-pages.csv',
 'Daily Files Not Duplicated/2nd Week 0

In [23]:
len(List_of_csv_file_paths)

43

In [24]:
df_concat2=None
for file in List_of_csv_file_paths:
    df=pd.read_csv(file)
    df_concat2=pd.concat([df_concat2,df],ignore_index=True)

In [25]:
df_concat2

Job Title  \
0                        LinkedIn Data Extractor   
1                           It Management Intern   
2                         SQL Database Developer   
3                Business Intelligence Developer   
4                     Web Developer - Internship   
...                                          ...   
15940                                        NaN   
15941                                        NaN   
15942            Information Security Specialist   
15943  FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
15944                        Community Associate   

                                                Job Link      Job ID  
0      https://www.linkedin.com/jobs/view/3727054368/...  3727054368  
1      https://www.linkedin.com/jobs/view/3723395858/...  3723395858  
2      https://www.linkedin.com/jobs/view/3727034665/...  3727034665  
3      https://www.linkedin.com/jobs/view/3727029273/...  3727029273  
4      https://www.linkedin.com/jobs/view/3727034696/...  3727034696  
...                                                  ...         ...  
15940                                                NaN  3767355025  
15941                                                NaN  3766809535  
15942  https://www.linkedin.com/jobs/view/3772530977/...  3772530977  
15943  https://www.linkedin.com/jobs/view/3771943212/...  3771943212  
15944  https://www.linkedin.com/jobs/view/3767944844/...  3767944844  

[15945 rows x 3 columns]

In [26]:
len(df_concat2)

15945

In [27]:
len(df_concat2['Job ID'].unique())

12321

In [29]:
duplicates_count=df_concat2.duplicated(subset=['Job ID']).sum()
duplicates_count

3624

In [30]:
len(df_concat2)-duplicates_count

12321